cpp/src/interop/detail/arrow_allocator.cpp

```diff
@@ -16,9 +16,40 @@
 
 #include <cudf/detail/interop.hpp>
 
+#include <memory>
+#include <sys/mman.h>
+#include <unistd.h>
+
 namespace cudf {
 namespace detail {
 
+/*
+  Enable Transparent Huge Pages (THP) for large (>4MB) allocations.
+  `buf` is returned untouched.
+  Enabling THP can improve performance of device-host memory transfers
+  significantly, see <https://github.com/rapidsai/cudf/pull/13914>.
+*/
+template <typename T>
+T enable_hugepage(T&& buf)
```

Comment:
could we have enable_hugepage just take a host_span? Do we really need to forward `std::unique_ptr<arrow::Buffer>` through the function?

Reply:
We don't need to but it means we can avoid an in-place function and have a simply return wrapper `return enable_hugepage(...)`.

Reply:
I still don't see the benefit, and I'm fine with that.